### ISSUE TITLE
Bug: Tree collapses when element selected beyond first InfiniteScroll load

### ISSUE THREAD

**Issue by nikolap** [dowjones/react-dropdown-tree-select#148]
Behaviour:
Clicking on any tree node after the first InfiniteScroll load (after you pass element 100) closes the tree. The item is correctly selected/deselected, however.

Intended behaviour:
Any element selection click (unless simple select true) should not hide the dropdown (i.e. tree). In this case, scrolling past the first InfiniteScroll load and clicking on any of the tree nodes should keep the tree and scroll position as it is.

To reproduce:
You can use the large-tree, for example (https://dowjones.github.io/react-dropdown-tree-select/#/story/large-tree)

1. Search for '1'
2. Scroll until the next batch of tree nodes loads
3. Click on any node
4. The node will be selected **but** the tree will be closed

Tested on Firefox 62.0b15 

Amazing job with this library, finding it very useful. Let me know if there's any clarification or debugging I can help with.

**Comment by nikolap** [comment]
Strange, Test 1 should have caused the bug:

![peek 2018-08-10 07-04](https://user-images.githubusercontent.com/4504265/43955063-a6f32b1a-9c6c-11e8-8015-f00cf1037f36.gif)

Anyways, I did some more digging and had it reproduced with the same version of Firefox (see gif above). So figured it must be a browser or addon thing and tried Chromium which worked just fine! So I went back and tried an old version of Firefox (v59), fresh and clean install. It had the same issue.

Long story short -- appears to be an issue on Firefox.

**Comment by mrchief** [comment]
Seems like something’s different in Firefox in regards to event handling. Inside click should not close the dropdown.

**Comment by nikolap** [comment]
Strangely, it doesn't close the dropdown *until* you scroll past the initial InfiniteScroll break.

![peek 2018-08-10 07-40](https://user-images.githubusercontent.com/4504265/43956078-bfe508d8-9c70-11e8-9e80-3efb36ff5a6f.gif)

**Comment by nikolap** [comment]
The closing/opening issue appeared to be due to the fallback path resolution in `getPath` (isOutsideClick.js). To simplify matters, Firefox has a composedPath function available and works as a good fallback before going to manually composing the path.

Note 1: I did not investigate why/where the fallback path resolution was failing in FF.
Note 2: There is a secondary bug (also browser specific) wherein the scroll position is reset incorrectly on Firefox, though correctly on Chromium.
